### PR TITLE
[Windows] Address CollectionView virtualization

### LINKED ISSUE #18639: [regression/8.0.0-rc.2.9530] Completely broken virtualization in CollectionView
### Description

The CollectionView loads all items from ItemSource, including those that are not within the view. This results in catastrophic performance in highly populated lists. The behavior was introduced in version 8.0.0-rc.2.9530. Tested on Windows.

In the sample the CollectionView's ItemSource receives Collection of 100 000 simple strings which results in progressive allocation of 4,4 GB and blocks main thread for several minutes.

### Steps to Reproduce

1. Load the repository and run the app.
2. Observe the raising memory usage in profiler/resource monitor.

### Link to public reproduction project repository

https://github.com/Vroomer/MAUI-CollectionView-broken-virtualization

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.0-rc.2.9511

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows SDK 10.0.19041.0

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

This PR some-what addresses virtualization being broken in `CollectionView` due to a lack of `MinHeight` on the items, the initial `MeasureOverride` call returning an infinite height, and creating the ListView items at the DataBinding-time. 

You can see the behavior in an isolated WinUI context here: https://github.com/Foda/ListViewVirtualization/tree/master

The fix for this issue is to just return a real-number value (32) when the initial call to `MeasureOverride` is done. It's not perfect, but very large lists will benefit from it.

Note: this behavior was regressed due to https://github.com/dotnet/maui/pull/18356

### Issues Fixed

Fixes #18639 #18510

### CODE REVIEW

File: src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs [hunk 1]
```diff
@@ -284,6 +284,10 @@ protected override WSize MeasureOverride(WSize availableSize)
 		{
 			if (_renderer == null)
 			{
+				// Make sure we supply a real number for height otherwise virtualization won't function
+				if (double.IsFinite(availableSize.Width) && !double.IsFinite(availableSize.Height))
```
Comment (Redth) [inline src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs:288]: Just want to make sure this is ok in CV scenarios where we have horizontal list scrolling set as the layout.

**Review by PureWeen — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Comment by enkaradag** [comment]
Hi,

I couldn't figure out where i shall use MeasureOverride to prevent CollectionView load all data at databinding

Here is my simple code

```
CollectionView view = new CollectionView() {HeightRequest=500};

///create 1000 rows of data here as ObservableCollection<MyDataClass> RowList

view.ItemTemplate = new DataTemplate(()=>
{
                Grid g = new Grid();
                Label lbl = new Label();
                lbl.SetBinding(Label.TextProperty, "FieldOfMyDataClass");
                g.Children.Add(lbl);
                return g;
}

view.ItemsSource=RowList

```

**Comment by enkaradag** [comment]
Hi,

i tried both MesureOverride of CollectionView and Grid. MeasureOverride of Grid is never calling, CollectionView's MeasureOverride is called with infinity value. Set it to an integer. But noway, it still attempts to loads all rows at databinding.

This issue is marked as fixed and closed. But i couldn't make it work for maui windows. Any help on this?

Or (I don't know the terminology) is this fix planned to be published later with SR1 or SR2? 

My older post https://github.com/dotnet/maui/issues/18913

Best Regards

**Comment by Elangovan4ever** [comment]
I also have the same issue in Window. Using 2000 items in collection view just with one label makes the app stuck and not respond. Using .NET 8 LTS. Please. If there is a workaround for time being, please let us know.

**Pull request merged**